### ISSUE TITLE
Using clousure type boolean

### ISSUE THREAD

**Issue by izi-dev** [epartment/nova-dependency-container#118]
`NovaDependencyContainer::make([
                KeyValue::make(__('Options'), 'option'),
            ])->dependsOn('type', function ($value) {
                   return $value == 1;
            } ),`

**Comment by ragingdave** [comment]
How exactly would you envision this work? Specifically around translating a PHP closure into something that would work on the JS side as well?

**Comment by izi-dev** [comment]
@ragingdave 
create this package meeting that need and in a slightly simpler way

https://novapackages.com/packages/izi-dev/nova-conditional-field

**Comment by izi-dev** [comment]
I did something similar, maybe this could help

https://github.com/izi-dev/nova-conditional-if

**Comment by wize-wiz** [comment]
@izi-dev yes, but the question remains, how to transform/translate the content of a PHP closure to JS.

**Comment by izi-dev** [comment]
If you have to make requests to the server, and with the NovaRequest we can access the fields with all their attributes, in the controller you can find how to do it.

**Comment by wize-wiz** [comment]
@izi-dev yes, for static group of fields, that would work. But how would the approach be when fields are added dynamically,  e.g with the flexible content package? Or even multiple nested dependency containers for that matter?

**Comment by izi-dev** [comment]
@wize-wiz 
If I understand, at the moment I did not find another way to solve it with the mentioned flexible content package and I imagine that it is the same with the other cases, this only works with simple fields I suppose, it will be necessary to continue investigating.

**Comment by wize-wiz** [comment]
@izi-dev well the problem is that any API call to a nova resource is a moment in time. It would be ridiculously complex to keep track of changes in the front-end and emulate any dynamic change in the backend with phantom/pseudo resources.

The real problem why, in my opinion, there won't be a solution to this problem, is that this package makes it possible to nest containers endlessly. Each change in the front-end never gets reflected with an API call to that resource. From a PHP point of view, the resource data is always static.

**Comment by izi-dev** [comment]
@wize-wiz 

I understand, the reason why I wanted this was because it only allows to send a single value so that the condition is fulfilled, and that is why I intended to do an anonymous function that would dynamically calculate the condition, but if we can send an array instead of an primitive value and so we have more option to assemble the possible answers so that a field is shown 


`-> dependsOn (“option”, [“select”, “checkbox”]`

**Comment by uandco** [comment]
I think "DependsOnContain" and "DependsOnNotContain" would be very useful and are easy to translate to JS (or add the ability to use regular expressions in DependsOn. I tried @izi-dev 's package and that works for my need but I use it on a text field (we need to show a password field only if the email field doesn't contain a certain domain) and that generates way too many API calls for my liking. Unfortunately I can't find another one that supports a "contains" type of condition.

**Comment by izi-dev** [comment]
@uandco If, in that case my package generates many calls to the api, maybe I will make an improvement on that, you can propose an issue, I am willing to help you.

**Comment by ragingdave** [comment]
@uandco DependsOnContain and DependsOnNotContain could be a good addition. Feel free to submit a PR
